internal/engine/build_and_deployer.go

```diff
@@ -66,15 +67,28 @@ func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context,
 	return BuildResult{}, lastErr
 }
 
+func isPermanentError(err error) bool {
```

Comment:
how are you defining "permanent error" here?

Reply:
added a comment!